force-app/main/default/classes/ElevateCreateBatchItemRequest.cls

```diff
@@ -28,50 +28,72 @@
  *     POSSIBILITY OF SUCH DAMAGE.
  */
 public with sharing class ElevateCreateBatchItemRequest {
-    //TODO: If we choose to support ACH we will need to explicitly pass paymentMethodType or the callout will fail
 
     public ElevateBatchItemType batchItemType;
     public CommitmentInfo commitmentInfo;
     public PurchaseInfo purchaseInfo;
 
-    public ElevateCreateBatchItemRequest(ElevateCreateBatchItemRequestDTO batchItemRequestDTO) {
+    public ElevateCreateBatchItemRequest(ElevateCreateBatchItemRequestDTO batchItemRequestDTO, String gatewayOverride) {
         if (batchItemRequestDTO.schedule == null || batchItemRequestDTO.schedule.isEmpty()) {
             batchItemType = ElevateBatchItemType.ONE_TIME;
         } else {
             batchItemType = ElevateBatchItemType.COMMITMENT;
         }
 
-        buildRequestBody(batchItemRequestDTO);
+        buildRequestBody(batchItemRequestDTO, gatewayOverride);
     }
 
-    private void buildRequestBody(ElevateCreateBatchItemRequestDTO batchItemRequestDTO) {
+    private void buildRequestBody(ElevateCreateBatchItemRequestDTO batchItemRequestDTO, String gatewayOverride) {
         PS_IntegrationServiceConfig.Service configService = new PS_IntegrationServiceConfig.Service();
 
         if (batchItemType == ElevateBatchItemType.COMMITMENT) {
-            buildCommitmentInfoBody(batchItemRequestDTO, configService);
+            buildCommitmentInfoBody(batchItemRequestDTO, configService, gatewayOverride);
         } else {
-            buildPurchaseInfoBody(batchItemRequestDTO, configService);
+            buildPurchaseInfoBody(batchItemRequestDTO, configService, gatewayOverride);
         }
     }
 
     private void buildPurchaseInfoBody(ElevateCreateBatchItemRequestDTO batchItemRequestDTO,
-                                       PS_IntegrationServiceConfig.Service configService) {
+                                       PS_IntegrationServiceConfig.Service configService,
+                                       String gatewayOverride) {
         purchaseInfo = new ElevateCreateBatchItemRequest.PurchaseInfo();
         purchaseInfo.merchantId = configService.getMerchantIds();
-        purchaseInfo.gatewayId = configService.getGatewayIds();
+        if (String.isNotBlank(gatewayOverride)) {
```

Comment:
Might not be worth it, but could the `configService` be passed the `gatewayOverride` and then move the if/else logic into the service itself?